### DISCUSSION TITLE
Inconsistent behavior with related units

### DISCUSSION THREAD

**Discussion by simoami** [josdejong/mathjs#3182] [Q&A]
Hello, I came across 2 different issues pertaining to units and I'm not sure if it's just a usage or configuration issue:

Issue 1:
I'm not sure why related units cause issues and what are the best practices in dealing with expressions involving related units
```
20 degC in degF // 68 degF (good 👍)
20 degC == 68 degF // false (unexpected 🤔)
20 degC == 36 degF  // true  (unexpected 🤔 I pinpointed the number 36 from trying several ranges)
20 degC in degF == 68 degF // true  (good 👍)
unit(20,degC) == unit(68,degF) // false (unexpected 🤔)
unit(20,degC) == unit(36,degF) // true (unexpected 🤔)
equal(unit(20,degC),unit(68,degF)) // false (unexpected 🤔)
20degC + 68degF in degF // 136 degF (good 👍)
```
Issue 2:
automatic unit conversion is nice but it's also inconsistent for my needs. I have an API that runs MathJS expressions and returns an object containing the following props:
- output: the raw evaluation output
- value: the value without unit. So we can apply number formatting and use it in analytics. obtained with `unit(output).toNumeric()`
- unit: the unit portion, so we can apply specific styling for the unit. obtained with `unit(output).formatUnits()`

Example 1: `bmi(w,h) = w / h ^ 2; bmi(70kg, 1.5m)`
- output: 31.11111111111111 kg / m^2
- value: 31.11111111111111
- unit: kg / m^2

Example 2: `2.3456 kg * 3.5 m/s^2`
- output: 8.21 N // notice the unexpected unit conversion to a Newton
- value: 8.21
- unit: (kg m) / s^2 // notice how it returns the non modified unit instead of N

Why are the output units inconsistent with the output of `unit.formatUnits()`. Is there a way for the raw output to return `8.21 (kg m) / s^2`.
FYI, I tried `toSI()` and it is more consistent. However it doesn't allow for output to be shown in requested units. Example: `(20degC to defF).toSI()` returns `293.15 K` even though I requested Fahrenheit and Kelvin is not the norm in certain applications.

**Comment by josdejong** [comment]
For reference, here is you question about unit systems: #3183

About temperatures, see:

- https://github.com/josdejong/mathjs/issues/2518
- https://github.com/josdejong/mathjs/issues/2499

About unit conversion: there is some logic that tries to return the result in as simple a unit as possible. In most of the cases that works very well, but like in your example you want more control. The current options for that are limited, see #3183. A hack to prevent the unit from being simplified to for example `N` is to set the property `.skipAutomaticSimplification` to `true` and only after that call `.toString()` on the unit (this is not a documented feature). The `toSI()` method will indeed try to stick to SI units like kelvin and meter over farenheit and inches.
